SAS Url "st", "se" must contain full date-time spec

BorisB2015 opened the pull request:
If one generates SAS Url using Azure CLI 2.0, it is possible to generate _valid_ SAS Url with date-only specification.

`az storage container generate-sas --connection-string $conn --name vhds --permissions rl --start '2017-08-01' --expiry '2017-08-20'`

This will trip AzCopy on later stages (it is a bug in AzCopy - reported and acknowledged on AzCopy doc page) while the URL is correct. Azure CLI 1.0 (azure command) always added time into "st", "se" even if not specified on the parameter.

It is required to use full date time like this:

`az storage container generate-sas --connection-string $conn --name vhds --permissions rl --start '2017-08-01T00:00:00Z' --expiry '2017-08-20T00:00:00Z'`

Added respective comments in CLI section and entry in SAS Url troubleshooting table.

azurecla commented:
Hi __@ahmetalpbalkan__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ahmetb). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

squillace commented:
Excellent, thanks. :-)

BorisB2015 committed:
SAS Url "st", "se" must contain full date-time spec

If one generates SAS Url using Azure CLI 2.0, it is possible to generate _valid_ SAS Url with date-only specification.

`az storage container generate-sas --connection-string $conn --name vhds --permissions rl --start '2017-08-01' --expiry '2017-08-20'`

This will trip AzCopy on later stages (it is a bug in AzCopy - reported and acknowledged on AzCopy doc page) while the URL is correct. Azure CLI 1.0 (azure command) always added time into "st", "se" even if not specified on the parameter.

It is required to use full date time like this:

`az storage container generate-sas --connection-string $conn --name vhds --permissions rl --start '2017-08-01T00:00:00Z' --expiry '2017-08-20T00:00:00Z'`

Added respective comments in CLI section and entry in SAS Url troubleshooting table.

GitHubber17 committed:
Update marketplace-publishing-vm-image-creation.md
